### ISSUE TITLE
How to train a localized dialect version of Lora?

### ISSUE THREAD

**Issue by zh584521** [koda-dernet/Side-Step#49]
Hello, I want to train a Chengdu dialect version of a Lora model. Previously, I trained it for 800 epochs using 20 data samples, but the output during testing didn’t include the dialect's tone, and the generated speech sounds like Mandarin. How should I proceed? I’d like to ask if this is achievable.

**Comment by koda-dernet** [comment]
It *should* be achievable but it depends on a lot of variables. The model you trained on, how many steps you trained for, what settings, how noticeable it is on your training data, how long your captions are, whether your lyrics are well written.

If you can provide me that information I could help you out with some advice

**Comment by zh584521** [comment]
Below is my training data format; I used 20 data points to train LoRa.

Metadata：
{
  "caption": "Classic Cantonese song, gentle and sentimental, soft vocal, deep emotion, nostalgic and touching.",
  "bpm": 72,
  "keyscale": "F major",
  "timesignature": "4/4",
  "genre": "cantonese, cantopop, classic, male vocal, sentimental, gentle, emotional, nostalgic"
}

lyrics：

[Verse]
我劝你早点归去
你说你不想归去
只叫我抱着你
悠悠海风轻轻吹
冷却了野火堆

[Verse]
我看见伤心的你
你说我怎舍得去
哭态也绝美
如何止哭
只得轻吻你发边

[Chorus]
让风继续吹
不忍远离
心里极渴望
希望留下伴着你

[Chorus]
风继续吹
不忍远离
心里亦有泪不愿流泪望着你

[Verse]
过去多少快乐记忆
何妨与你一起去追
要将忧郁苦痛洗去
柔情蜜意我愿记取

[Verse]
要强忍离情泪
未许它向下垂
愁如锁
眉头聚
别离泪始终要下垂

[Chorus]
我已令你快乐
你也令我痴痴醉
你已在我心
不必再问记着谁
留住眼内每滴泪
为何仍断续流默默垂

[Verse]
我劝你早点归去
你说你不想归去
只叫我抱着你
悠悠海风轻轻吹
冷却了野火堆

[Verse]
我看见伤心的你
你叫我怎舍得去
哭态也绝美
如何止哭
只得轻吻你发边

[Chorus]
让风继续吹
不忍远离
心里极渴望
希望留下伴着你

[Chorus]
风继续吹
不忍远离
心里亦有泪不愿流泪望着你

[Verse]
过去多少快乐记忆
何妨与你一起去追
要将忧郁苦痛洗去
柔情蜜意我愿记取

[Verse]
要强忍离情泪
未许它向下垂
愁如锁
眉头聚
别离泪始终要下垂

[Chorus]
我已令你快乐
你也令我痴痴醉
你已在我心
不必再问记着谁
留住眼内每滴泪
为何仍断续流默默垂

[Outro]
为何仍断续流默默垂
为何仍断续流默默垂

<img width="1900" height="918" alt="Image" src="https://github.com/user-attachments/assets/4701b349-fd0d-4e5c-a58e-1390f35f03e6" />[](url)

**Comment by zh584521** [comment]
> Below is my training data format; I used 20 data points to train LoRa.
> 
> Metadata： { "caption": "Classic Cantonese song, gentle and sentimental, soft vocal, deep emotion, nostalgic and touching.", "bpm": 72, "keyscale": "F major", "timesignature": "4/4", "genre": "cantonese, cantopop, classic, male vocal, sentimental, gentle, emotional, nostalgic" }
> 
> lyrics：
> 
> [Verse] 我劝你早点归去 你说你不想归去 只叫我抱着你 悠悠海风轻轻吹 冷却了野火堆
> 
> [Verse] 我看见伤心的你 你说我怎舍得去 哭态也绝美 如何止哭 只得轻吻你发边
> 
> [Chorus] 让风继续吹 不忍远离 心里极渴望 希望留下伴着你
> 
> [Chorus] 风继续吹 不忍远离 心里亦有泪不愿流泪望着你
> 
> [Verse] 过去多少快乐记忆 何妨与你一起去追 要将忧郁苦痛洗去 柔情蜜意我愿记取
> 
> [Verse] 要强忍离情泪 未许它向下垂 愁如锁 眉头聚 别离泪始终要下垂
> 
> [Chorus] 我已令你快乐 你也令我痴痴醉 你已在我心 不必再问记着谁 留住眼内每滴泪 为何仍断续流默默垂
> 
> [Verse] 我劝你早点归去 你说你不想归去 只叫我抱着你 悠悠海风轻轻吹 冷却了野火堆
> 
> [Verse] 我看见伤心的你 你叫我怎舍得去 哭态也绝美 如何止哭 只得轻吻你发边
> 
> [Chorus] 让风继续吹 不忍远离 心里极渴望 希望留下伴着你
> 
> [Chorus] 风继续吹 不忍远离 心里亦有泪不愿流泪望着你
> 
> [Verse] 过去多少快乐记忆 何妨与你一起去追 要将忧郁苦痛洗去 柔情蜜意我愿记取
> 
> [Verse] 要强忍离情泪 未许它向下垂 愁如锁 眉头聚 别离泪始终要下垂
> 
> [Chorus] 我已令你快乐 你也令我痴痴醉 你已在我心 不必再问记着谁 留住眼内每滴泪 为何仍断续流默默垂
> 
> [Outro] 为何仍断续流默默垂 为何仍断续流默默垂
> 
> <img alt="Image" width="1900" height="918" src="https://private-user-images.githubusercontent.com/38693640/579158583-4701b349-fd0d-4e5c-a58e-1390f35f03e6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzYzMzYxOTEsIm5iZiI6MTc3NjMzNTg5MSwicGF0aCI6Ii8zODY5MzY0MC81NzkxNTg1ODMtNDcwMWIzNDktZmQwZC00ZTVjLWE1OGUtMTM5MGYzNWYwM2U2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDE2VDEwMzgxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiODJmMTJjNDMzNTZjNjA3ZjJkMGU2NmZkMzA4OWM1NmQ5YWZkYzE3NGZkZmQxMzRhOTcwNjJlZGUxODE3YWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.dj-wyIwihw1QjWqgs2EuT4d6433MrdRQ4oMJwN3nMQ0">

@koda-dernet

**Comment by koda-dernet** [comment]
So it seems you tried the gradio training. Side-Step does things differently, but if you don't want to do the switch, a few things you can try in the Gradio trainer is

- Increase your rank to 32 or 64, and your alpha by 2x the rank
- Increase your Learning Rate to 3e-4 (currently 1e-4)

I would tell you to try more things, but the Gradio trainer is a bit limited, you can try either one of those two things above, or try both at the same time.

**Comment by zh584521** [comment]
> 看来你尝试了gradio训练。侧步做不同，但如果你不想切换，可以在Gradio器里尝试几件事
> 
> * 把你的等级提升到32或64，alpha等级提升2倍
> * 提高你的学习速度到3e-4（目前是1e-4）
> 
> 我本想建议你多试几种，但Gradio训练器有点限制，你可以尝试上面说的这两种，或者同时用两种。

ok, thank you. I'll give it a try. Is Side-Step better than the official Grado? Could you please give me your contact information?

**Comment by koda-dernet** [comment]
Not to toot my own horn, but yes, Side-Step is better than the official trainer; mostly because the official trainer is made for ACE-Step 1.5 Turbo [and not so much the other variants].

On top of that, I give a lot (seriously, a lot) of features so you can arrange and optimize your training run in so many ways.

You also get a dedicated, good looking GUI for your local hardware, an intuitive terminal Wizard UI if you prefer that, and a CLI optimized for cloud deployments.

You can contact me through Discord in the ACE-Step server, you will find me easily there as dernet

Otherwise you can also contact me through my email gadna166@gmail.com

**Comment by zh584521** [comment]
> Discord 在 ACE-Step 服务器上联系我，我的用户名是 dern

ok, thank you. I have indeed improved based on the training you suggested, but the background noise is a bit loud.
The following was generated during testing. Could you please provide feedback? Thank you.
[add66d9a-93bc-445c-ae14-e0639711e5b4(1).mp3](https://github.com/user-attachments/files/26811882/add66d9a-93bc-445c-ae14-e0639711e5b4.1.mp3)

[fbd3d038-9ec0-51fe-a603-b8861f879aa1(1).mp3](https://github.com/user-attachments/files/26811885/fbd3d038-9ec0-51fe-a603-b8861f879aa1.1.mp3)
